bids_prov/spm/spm_parser.py

```diff
@@ -187,54 +190,74 @@ def get_entities_from_ext_config(conf_dic: dict, activity_name: str, activity_id
     in records_activities, update id in activity["used"], and return output_entity
 
     Parameters
     ----------
-    records_activities : list of the previous recorded activities
-    activity :  current activity
+    activities : All previous activities
     right : current right end of line
-    verbose : True to have more verbosity
 
     Returns
     -------
     output_entity : it is the generated entity with the  corresponding closest activity
 
     """
-
-    dependency = re.search(conf.DEPENDENCY_REGEX, right, re.IGNORECASE)  # cfg_dep\(['"]([^'"]*)['"]\,.*
-    # check if the line call cfg_dep and retrieve the first parameter retrieve all digits between parenthesis
-    output_entity = dict()
     # and retrieve the first parameter,  all digits between parenthesis
-    dep_number = re.search(r"{(\d+)}", right)
-    # retrieve name of the output_entity
-    parts = dependency.group(1).split(": ")
-    # if right = "cfg_dep('Move/Delete Files: Moved/Copied Files', substruct('.',...));"
-    # return : ['Move/Delete Files', 'Moved/Copied Files'
-    for act in records_activities:
-
+    dep_number = re.search(r"{(\d+)}", right) # retrieve name of the output_entity
+    closest_activity = None
+    for act in activities:
         if act["label"].endswith(dep_number.group(1)):
             closest_activity = act
-            if verbose:
-                print(f"closest_activity : {closest_activity}")
-
-            output_id = next((entity["@id"] for entity in records["prov:Entity"]
-                              if parts[-1] == entity["label"] and entity["generatedBy"] == closest_activity["@id"]),
-                             "urn:" + get_id())
-
-            # adds to the current activity the fact that it has used the previous entity
-            activity["used"].append(output_id)
-            output_entity = {
-                "@id": output_id,
-                "label": label_mapping(parts[-1], "spm/spm_labels.json"),
-                # "prov:atLocation": TODO
-                "generatedBy": closest_activity["@id"],
-            }
+            break
+
+    return closest_activity
 
-    return output_entity
+def find_output_id_from_closest(closest_activity,records):
+    for entity in records["prov:Entity"]:
+        if "generatedBy" in entity:
+            if entity["generatedBy"] == closest_activity["@id"]: # entity["label"] == parts[-1]
+                output_id = entity["@id"]
+                break
+    else:
+        output_id = "urn:" + get_id()
+    # output_id = next(
+    #     (entity["@id"] for entity in records["prov:Entity"]
+    #                   if parts[-1] == entity["label"] and entity["generatedBy"] == closest_activity["@id"]
+    #     ), "urn:" + get_id()
+    #     ) # find entity id generated by closest activity, if None new get_id
+
+    return output_id
+
+def mapping_label_entity(key, filename):
+    mapping = {
```

Comment:
Can you add a comment to document what this function does.

Reply:
Done